kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -148,86 +110,129 @@ object Result:
       *
       * @param either
       *   The Either to convert
-      * @tparam E
-      *   The type of the error (Left value)
-      * @tparam A
-      *   The type of the successful value (Right value)
       * @return
       *   A Result equivalent to the input Either
       */
     def fromEither[E, A](either: Either[E, A]): Result[E, A] =
-        either.fold(fail, success)
+        either.fold(fail, succeed)
 
     /** Converts a Try to a Result.
       *
       * @param t
       *   The Try to convert
-      * @tparam A
-      *   The type of the successful value
       * @return
       *   A Result equivalent to the input Try
       */
-    def fromTry[A](t: Try[A]): Result[Throwable, A] =
-        t.fold(fail, success)
+    def fromTry[A](t: Try[A]): Result[Throwable, A] = t.fold(fail, succeed)
+
+    /** Collects a sequence of Results into a single Result containing a sequence of successful values.
+      *
+      * @param seq
+      *   The sequence of Results to collect
+      * @return
+      *   A Result containing either a sequence of successful values or the first encountered error
+      */
+    def collect[E, A](seq: Seq[Result[E, A]]): Result[E, Seq[A]] =
+        def loop(remaining: Seq[Result[E, A]], acc: Chunk[A]): Result[E, Chunk[A]] =
+            remaining match
+                case (head: Result[E, A]) +: tail =>
+                    head.flatMap(value => loop(tail, acc.append(value)))
+                case Seq() => Success(acc)
+
+        try loop(seq, Chunk.empty[A])
+        catch
+            case ex => Panic(ex)
```

Comment:
This catch was missing here. It's covered by the tests now

Reply:
in case the traversal fails: https://github.com/getkyo/kyo/pull/1023/files/348aaef3741c209f385fa93944c38bb4168c5569#diff-0e34607ea57bda29f4f9c1d41662285cf847a5943ee369dc48afeaf1f204dd36R1253